src/bsmHandler.cpp

```diff
@@ -376,110 +375,27 @@ bool BSMHandler::process( const std::string& bsm_json ) {
-
-        // get partII data
-        rapidjson::Value& partII = data["partII"];
-
-        // for each field
-        for (std::string member : rpm.getFields()) {
-            if (debug) {
-                bool psuccess = false;
-                isMemberPresent(partII, member, psuccess);
-                std::cout << "Is the '" << member << "' member present... Before redaction? " << psuccess;
-            }
-
-            // redact field
-            bool success = false;
-            findAndRemoveAllInstancesOfMember(partII, member.c_str(), success);
-            if (success) {
-                numMembersRedacted++;
-            }
-
-            if (debug) {
-                bool psuccess = false;
-                isMemberPresent(partII, member, psuccess);
-                std::cout << "----- After redaction? " << psuccess << std::endl;
-            }
+void BSMHandler::handleGeneralRedaction(rapidjson::Document& document) {
+    if (is_active<kGeneralRedactFlag>()) {
+        for (std::string memberPath : rpm.getFields()) {
+            bool memberRedacted = false;
+            rapidjsonRedactor.redactMemberByPath(document, memberPath.c_str(), memberRedacted);
         }
-        if (debug) { std::cout << "Members redacted: " << numMembersRedacted << std::endl; }
-        std::string partIIString = convertRapidjsonValueToString(partII);
-        bsm_.set_partII(partIIString);
-    }
-}
 
-/**
- * @brief Recursively search for a member in a value and remove all instances of it it.
- * 
- * @param value The value to begin searching.
- * @param member The member to remove all instances of.
- * @param success The flag that the caller can check upon return to see if the operation was successful.
- */
-void BSMHandler::findAndRemoveAllInstancesOfMember(rapidjson::Value& value, std::string member, bool& success) {
-    static const char* kTypeNames[] = { "Null", "False", "True", "Object", "Array", "String", "Number" };
-    if (value.IsObject()) {
-        while (value.HasMember(member.c_str())) {
-            value.RemoveMember(member.c_str());
-            success = true;
-        }
-        for (auto& m : value.GetObject()) {
-            std::string type = kTypeNames[m.value.GetType()];
-            if (type == "Object" || type == "Array") {
-                std::string name = m.name.GetString();
-                auto& v = value[name.c_str()];
-                findAndRemoveAllInstancesOfMember(v, member, success);
-            }
-        }
-    }
-    else if (value.IsArray()) {
-        for (auto& m : value.GetArray()) {
-            std::string type = kTypeNames[m.GetType()];
-            if (type == "Object" || type == "Array") {
-                findAndRemoveAllInstancesOfMember(m, member, success);
-            }
-        }
-    }
-}
+        // attempt to store the redacted coreData and partII in the BSM object
+        rapidjson::Value& payload = document["payload"];
```

Comment:
Would it make sense to just check something like `document["payload"]["data"].HasMember("coreData")` below rather than declaring the payload and data objects? I'm not sure how the C++ code handles the memory since we're creating a reference object here but my assumption is that two new blocks of memory are set aside to store those new variables.